nixos/platform/shell.nix

```diff
@@ -64,15 +67,16 @@ in
     );
 
     users.motd = let
-      has_release_name = parameters ? release_name;
+      inherit (config.flyingcircus.platform) release;
+      isStableRelease = release != {};
     in
     ''
       Welcome to the Flying Circus!
 
       Status:     https://status.flyingcircus.io/
       Docs:       https://doc.flyingcircus.io/
-      ${ opt (parameters ? release_changelog ) ("ChangeLog:  " + parameters.release_changelog)}
-      Release:    ${ opt has_release_name "${parameters.release_name} (" + config.system.nixos.label + opt has_release_name ")"}
+      Release:    ${opt isStableRelease "${release.release_name} (" + config.system.nixos.label + opt isStableRelease ")"}
+      ${opt isStableRelease ("ChangeLog:  " + release.release_changelog)}
```

Comment:
This can look weird if there is no Changelog (that field is optional). We can of course say that all stable releases *should* have a changelog and failing to provide one is the core issue.

Reply:
I leave it like that to show that something we expect is actually missing.